From ceea63943d035770464e974058024a4a44d44355 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sun, 22 Dec 2024 13:26:25 +0200
Subject: [PATCH] tests: Stop hlr_auc_gw more cleanly

Use the new TERMINATE command through the socket to stop hlr_auc_gw
instead of depending on killall. There seemed to be some kind of race
condition with UML that could prevent cleanup previously.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 tests/hwsim/stop.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/hwsim/stop.sh b/tests/hwsim/stop.sh
index 22e24cd39..514993c1b 100755
--- a/tests/hwsim/stop.sh
+++ b/tests/hwsim/stop.sh
@@ -18,6 +18,12 @@ if grep -q hwsim0 /proc/net/dev; then
     sudo ip link set hwsim0 down
 fi
 
+if [ -e /tmp/hlr_auc_gw.sock ]; then
+    if which -s socat; then
+	echo TERMINATE | socat - UNIX-SENDTO:/tmp/hlr_auc_gw.sock
+	sleep 0.1
+    fi
+fi
 sudo killall -q hlr_auc_gw
 
 if [ "$RUNNING" = "yes" ]; then
-- 
2.43.0

